### DISCUSSION TITLE
代码生成器中的编辑功能，点击保存后无效

### DISCUSSION THREAD

**Discussion by oknixus** [slowlyo/owl-admin#90] [Anything]
![GIF 2024-4-1 18-36-49](https://github.com/slowlyo/owl-admin/assets/4592706/9b9006ed-3930-4c8a-a9b3-8fcce802fd7e)

**Comment by slowlyo** [comment]
可能是 Octane 的原因, 没有专门适配过

**Reply by oknixus** [reply]
有适配的计划吗？

**Reply by oknixus** [reply]
又是这种情况：在`AdminService.php`中的`public function update(){}`中，加了一行`Log::info('参数：', $data);`后，再次执行修改操作，就正常了

**Reply by oknixus** [reply]
还有一个问题是，修改`功能名称`或`菜单名称`，修改成功后，`admin_menus`中的`title`的值没有被更新，仍然是新建时候的值

**Reply by slowlyo** [reply]
后续会适配的

**Reply by oknixus** [reply]
会介意在这里一直发帖请教问题吗？比如，多用户的功能，遇到了个错误`Auth guard [biz] is not defined.`。已经解决了，但是或许应该反馈一下

**Reply by slowlyo** [reply]
这个不是框架内的问题吧

**Reply by oknixus** [reply]
可能不是。看了下生成的代码结构，在biz这个模块下，有一个config.php文件，已经包含了guard

**Reply by oknixus** [reply]
我切回fpm试试

**Reply by slowlyo** [reply]
config/admin.php 下没有注册模块吧

**Reply by oknixus** [reply]
严格按照文档https://doc.owladmin.com/#/guide/install进行安装的。
出现`Auth guard [biz] is not defined.`这个错误后，在`config/admin.php`中的guard里加了biz，然后就可以访问了。后面发现在`admin-modules\Biz`下还有一个`config.php`文件，打开一看，这里也是有guard信息的，里面就包含了biz。然后就删掉了在`config/admin.php`中guard里手动添加的biz，然后就再次出现了`Auth guard [biz] is not defined.`

新构建了一个php-fpm的镜像，访问 http://localhost:8004/biz#/system/admin_users 就没再出现`Auth guard [biz] is not defined.`这个错误了。但是，菜单全部都是英文的。而octane模式下的菜单，全是中文的。

另外就是，fpm模式下，页面加载非常慢，API基本都在4s左右。很好奇http://demo.owladmin.com/admin#/dashboard这个服务器配置和软件配置，为什么这么快

**Reply by oknixus** [reply]
![image](https://github.com/slowlyo/owl-admin/assets/4592706/78bee4ac-d34a-4b48-b2c7-058fe38f0ece)

**Reply by slowlyo** [reply]
正常情况下接口响应都是在200ms内的, 你这个这么慢, 是不是用的 wsl 然后代码放在本机?

**Reply by slowlyo** [reply]
菜单显示的问题   可能是lang目录没有权限

**Reply by oknixus** [reply]
太慢这个问题，可能和opcache有关，修改了配置，现在也快了很多
docker确实选了wsl

**Reply by oknixus** [reply]
有哪些产品在用这个框架，我是很喜欢这个框架的，真的太方便了。虽然用不了octane，不过我可以先在fpm模式下开发。现在想看看哪些产品在用，好评估一下是否适合当前项目

**Reply by slowlyo** [reply]
并没有共享出来的产品, 因为这是个后台框架, 看示例可以参考demo 或 官网

**Reply by oknixus** [reply]
建议开个帖子，让使用了该框架的产品，发出来

**Reply by wcz0** [reply]
不能看到后台的产品可以发吗. [yunshuiban](https://open.yunshuiban.com/index)

**Comment by oknixus** [comment]
相同的代码，同时跑`swoole+octane`容器和`fpm+opcache`容器
![image](https://github.com/slowlyo/owl-admin/assets/4592706/2beaa173-50c0-422d-a882-853599b7f1b5)
![image](https://github.com/slowlyo/owl-admin/assets/4592706/c4a755fb-1a0c-444f-96b3-696ff743415b)
